samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitor.java

```diff
@@ -147,6 +145,8 @@ private void markSweepTaskStore(File taskStoreDir) throws IOException {
     if (!offsetFile.exists()) {
       LOG.info("Deleting the task store : {}, since it has no offset file.", taskStorePath);
       FileUtils.deleteDirectory(taskStoreDir);
+      localStoreMonitorMetrics.diskSpaceFreedInBytes.inc(taskStoreDir.getTotalSpace());
```

Comment:
This was tested in my devbox(execution environment is on open-jdk). 

`Does getTotalSpace() return appropriate value?`

It depends upon the JDK/JVM implementation of File interface. In some implementations, the size of the File handle in bytes is read(along with other file attributes) when the contents are read/written(which is inline with the documentation for this interface). My sample test totally separate from this work fine with that.  

In some implementations, this total bytes value is fetched during each call of this API(no pre-fetch & storage during read/write call). 

 I'm moving this getTotalSpace() call before the delete operation.